stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2063,15 +2069,15 @@ extension Dictionary where Key : Equatable, Value : Equatable {
         let (key, value) = lhsNative.assertingGet(index)
         let optRhsValue: AnyObject? =
           rhsCocoa.maybeGet(_bridgeAnythingToObjectiveC(key))
-        // TODO: swift-3-indexing-model: change 'if' into 'guard'.
-        if let rhsValue = optRhsValue {
-          if value == _forceBridgeFromObjectiveC(rhsValue, Value.self) {
-            lhsNative.formIndex(after: &index)
-            continue
-          }
+
+        guard let rhsValue = optRhsValue,
+          value == _forceBridgeFromObjectiveC(rhsValue, Value.self)
+        else {
+          return false
         }
-        index = lhsNative.index(after: index)
-        return false
+
```

Comment:
FYI, in general I am wary of `guard` and these compound conditions; IMO they often don't actually improve readability of the code, I find.  I don't have a strong opinion in this case, but I ask that when you do use `guard` that you convince yourself that the code has improved.